readthedocs/core/views/serve.py

```diff
@@ -401,15 +400,18 @@ def changefreqs_generator():
 
         if project.translations.exists():
             for translation in project.translations.all():
-                href = project.get_docs_url(
-                    version_slug=version.slug,
-                    lang_slug=translation.language,
-                    private=version.privacy_level == constants.PRIVATE,
-                )
-                element['languages'].append({
-                    'hreflang': translation.language,
-                    'href': href,
-                })
+                translation_versions = translation.versions.public(
+                    ).values_list('slug', flat=True)
+                if version.slug in translation_versions:
+                    href = project.get_docs_url(
+                        version_slug=version.slug,
+                        lang_slug=translation.language,
+                        private=version.privacy_level == constants.PRIVATE,
```

Comment:
This can be `False` directly, we are already filtering by `.public`.

Reply:
okay :+1: